### PR TITLE
Server bugfix 019

### PR DESCRIPTION
**Pull request by khc196**

### CODE REVIEW

File: server/OpenCafeFinder.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*
+
+# input : list organized by python dictionary
+# output : print (insert success or fail)
+
+import json
+import pymysql
+from datetime import datetime
+import pytz
```
Comment (shberry) [inline server/OpenCafeFinder.py]: group imports together

File: server/OpenCafeFinder.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+# connect to DB
+def getConnection():
+    return pymysql.connect(host='localhost',
+                            user='gaeul',
+                            password='alpha',
+                            db='CAFE',
+                            charset='utf8mb4')
+
+# inquiry data from connected DB
```
Comment (shberry) [inline server/OpenCafeFinder.py]: 'Fetch' is more commonly used :) FYI.

File: server/OpenCafeFinder.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                            user='gaeul',
+                            password='alpha',
+                            db='CAFE',
+                            charset='utf8mb4')
+
+# inquiry data from connected DB 
+def getTable():
+    conn = getConnection()
+    try:
+        with conn.cursor(pymysql.cursors.DictCursor) as cursor :
```
Comment (shberry) [inline server/OpenCafeFinder.py]: Should we log errors when connection goes wrong somehow?

File: server/OpenCafeFinder.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        return cafe_list
+    finally:
+        conn.close()
+
+# get information about date
+def getDate() :
+    now_day = datetime.today().weekday()
+    utc_now = pytz.utc.localize(datetime.utcnow())
+    now = utc_now.astimezone(pytz.timezone("America/Los_Angeles"))
+    #now = datetime.now()
```
Comment (shberry) [inline server/OpenCafeFinder.py]: remove commented-out code chunks

File: server/OpenCafeFinder.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    #now = localtz.localize(now)
+    now_hour = now.hour
+    now_minute = now.minute
+    return now_day, now_hour, now_minute #return type is tuple 
+
+# find out opening cafe
```
Comment (shberry) [inline server/OpenCafeFinder.py]: 'Find currently open cafe'?

File: server/OpenCafeFinder.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            open_minute = int(cafe[day_to_find+'_open_hours'][8] + cafe[day_to_find+'_open_hours'][9])
+            close_hour = int(cafe[day_to_find+'_open_hours'][15] + cafe[day_to_find+'_open_hours'][16])
+            close_minute = int(cafe[day_to_find+'_open_hours'][17] + cafe[day_to_find+'_open_hours'][18])
+            if (((open_hour == now_hour and open_minute <= now_minute) or (open_hour < now_hour)) and \
+                ((close_hour == now_hour and close_minute >= now_minute) or (close_hour > now_hour))) :
+                open_cafe_list.append(cafe)
+    return open_cafe_list
+
+if __name__ == "__main__":
+    cafe_list = getTable()
```
Comment (shberry) [inline server/OpenCafeFinder.py]: Why do we get the whole table? Can we query the table with conditions mentioned in line 56, 57?

File: server/webhook.py [hunk 7]
```diff
@@ -102,7 +102,7 @@ def makeWebhookResult(data):
 
     # insert mapping data to redis table
     redis_obj.set(token_generated, json.dumps(data))
-    redis_obj.expire(token_generated, 60)
+    #redis_obj.expire(token_generated, 60)
```
Comment (shberry) [inline server/webhook.py:104]: remove the commented out code

File: server/webhook.py [hunk 8]
```diff
@@ -113,4 +113,9 @@ def makeWebhookResult(data):
 def run() :
     port = int(os.getenv('PORT', 5002))
     print("Starting Webhook Server on port %d" % port)
-    app.run(debug=False, port=port, host='0.0.0.0')
\ No newline at end of file
+    app.run(debug=False, port=port, host='0.0.0.0')
+
+if __name__ == '__main__' : 
+    port = int(os.getenv('PORT', 5002))
+    print("Starting Webhook Server on port %d" % port)
```
Comment (shberry) [inline server/webhook.py]: If this is for testing purpose, remove.

File: server/webhook.py [hunk 9]
```diff
@@ -113,4 +113,9 @@ def makeWebhookResult(data):
...
     port = int(os.getenv('PORT', 5002))
     print("Starting Webhook Server on port %d" % port)
-    app.run(debug=False, port=port, host='0.0.0.0')
\ No newline at end of file
+    app.run(debug=False, port=port, host='0.0.0.0')
+
+if __name__ == '__main__' : 
+    port = int(os.getenv('PORT', 5002))
+    print("Starting Webhook Server on port %d" % port)
+    app.run(debug=False, port=port, host='0.0.0.0')
```
Comment (shberry) [inline server/webhook.py]: Do we use host 0.0.0.0 for production code as well?

File: server/webserver.py [hunk 10]
```diff
@@ -73,13 +73,12 @@ def getCafes(token, latitude, longitude) :
...
-    data = getDatafromDB(query)
-    cafe_list = json.loads(data)
-    # filter cafes by radius
-    result = filterbyradius(cafe_list, latitude, longitude)
+    cafe_list = getDatafromDB(query)
+    open_cafe_list = findOpenCafes(filterbyradius(cafe_list, latitude, longitude))
     
     response = app.response_class(
-        response=result,
+        #response=result,
```
Comment (shberry) [inline server/webserver.py]: remove

File: server/webserver.py [hunk 11]
```diff
@@ -93,4 +92,9 @@ def index(token) :
 def run() :
     port = int(os.getenv('PORT', 5000))
     print("Starting Web Server on port %d" % port)
-    app.run(debug=False, port=port, host='0.0.0.0', ssl_context=('cert.pem', 'key.pem'))
\ No newline at end of file
+    app.run(debug=False, port=port, host='0.0.0.0', ssl_context=('cert.pem', 'key.pem'))
+
+if __name__ == '__main__' :
+    port = int(os.getenv('PORT', 5000))
+    print("Starting Web Server on port %d" % port)
```
Comment (shberry) [inline server/webserver.py]: Same here. If this is for testing, remove.

**Review by shberry — COMMENTED** [review]
Did a quick round. Thanks!

**Comment by GaeulG** [comment]
cool

**Review by GaeulG — APPROVED** [review]

**Pull request merged**